optimism-contract-deployer/Dockerfile

```diff
@@ -12,47 +12,34 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     curl \
     gcc \
     g++ \
-    python3 \
-    python3-pip \
     vim \
+    curl \
     build-essential \
     libusb-1.0-0-dev \
     libssl-dev \
     ca-certificates \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
 
+RUN GO_VERSION=$(curl -s https://raw.githubusercontent.com/ethereum-optimism/optimism/develop/versions.json | jq -r '.go') && curl -sL https://go.dev/dl/go$GO_VERSION.linux-amd64.tar.gz -o go$GO_VERSION.linux-amd64.tar.gz &&  tar -C /usr/local/ -xzvf go$GO_VERSION.linux-amd64.tar.gz && rm go$GO_VERSION.linux-amd64.tar.gz
+ENV GOPATH=/go
+ENV PATH=/usr/local/go/bin:$GOPATH/bin:$PATH
 
-# Install Go from the official golang image
-# Revert to golang:alpine once https://github.com/ethereum-optimism/optimism/pull/11494 is merged
-COPY --from=golang:1.22.6 /usr/local/go/ /usr/local/go/
-ENV PATH="/usr/local/go/bin:${PATH}"
-
-# Install web3 cli
-RUN curl -LSs https://raw.githubusercontent.com/gochain/web3/master/install.sh | sh
+# Clone the Optimism monorepo and build the `op-node` binary for L2 genesis generation.
+RUN git clone https://github.com/ethereum-optimism/optimism.git && \
+    cd optimism && \
+    git checkout develop && \
+    git pull origin develop && \
+    cd op-node && \
+    make
 
-# Install Rust and Foundry
+# Install foundry
 RUN curl -L https://foundry.paradigm.xyz | bash
 ENV PATH="/root/.foundry/bin:${PATH}"
-RUN foundryup
+RUN FOUNDRY_VERSION=$(curl -s https://raw.githubusercontent.com/ethereum-optimism/optimism/develop/versions.json | jq -r '.foundry') && foundryup -v nightly-$FOUNDRY_VERSION
 
-# RUN git clone --recursive https://github.com/ethereum-optimism/optimism.git && \
-#     cd optimism && \
-#     git submodule update && \
-#     git checkout develop && \
-#     git pull origin develop && \
-#     git fetch --all && \
-#     cd op-node && \
-#     make
-
-RUN git clone --recursive https://github.com/barnabasbusa/optimism.git && \
```

Comment:
So, you're optimism now? :D
What's the reason to switch repos? 
can't see anything in your repo that isn't in the upstream repo?

Reply:
That is getting removed. Now we are going back to using upstream optimism repo. I was just testing my PR.